cmd/jaeger/internal/extension/jaegerstorage/extension.go

```diff
@@ -126,7 +126,7 @@ func (s *storageExt) Start(_ context.Context, _ component.Host) error {
 			factory, err = badger.NewFactoryWithConfig(*cfg.Badger, mf, s.telset.Logger)
 		case cfg.GRPC != nil:
 			//nolint: contextcheck
-			factory, err = grpc.NewFactoryWithConfig(*cfg.GRPC, mf, s.telset.Logger)
+			factory, err = grpc.NewFactoryWithConfig(*cfg.GRPC, mf, s.telset.Logger, grpc.NewHost())
```

Comment:
you don't need to create a new host, its passed into this method. Just change the `_` in the function signature to be `host` and pass that in here.